config/routes.rb

```diff
@@ -3,10 +3,16 @@
   resources :companies do
     resources :shops do
       resources :employees
+      resources :shop_slots, except: [:index, :show]
+      get :index_appointment
+      put :change_status
     end
   end
 
-  resources :users
+  resources :users do
+    resources :appointments, except: [:delete, :index]
```

Comment:
delete should be possible. users should be able to delete own appointments if it is still in booked